.github/workflows/terragrunt-apply-production.yml

```diff
@@ -82,7 +82,7 @@ jobs:
 
   terragrunt-apply-all-modules:
     needs: build-tag-push-lambda-images
-    if: always() && (needs.build-tag-push-lambda-images.result == 'success' || needs.build-tag-push-lambda-images.result == 'skipped')
+    if: ${{ !failure() && !cancelled() }}
```

Comment:
Alright so yeah according to https://stackoverflow.com/questions/69354003/github-action-job-fire-when-previous-job-skipped#comment138343758_77066140 we would not need the `always` to force the job execution even if the previous one was skipped.
Let's try it :)